icann-rdap-client/src/gtld/entity.rs

```diff
@@ -0,0 +1,225 @@
+                        "url" => url = property[3].as_str().unwrap_or("").to_string(),
+                        "org" => org = property[3].as_str().unwrap_or("").to_string(),
+                        "adr" => {
+                            if let Some(address_components) = property[3].as_array() {
+                                adr = format_address_with_label(params, address_components);
+                            }
+                        }
+                        _ => {}
+                    }
+                }
+            }
+        }
+    }
+
+    RoleInfo {
+        name,
+        org,
+        url,
+        adr,
+    }
+}
+
+fn append_abuse_contact_info(entity: &Entity, front_formatted_data: &mut String) {
+    if let Some(entities) = &entity.object_common.entities {
+        for entity in entities {
+            if let Some(roles) = &entity.roles {
+                for role in roles {
+                    if role.as_str() == "abuse" {
+                        if let Some(vcard_array) = &entity.vcard_array {
+                            if let Some(properties) = vcard_array[1].as_array() {
+                                for property in properties {
+                                    if let Some(property) = property.as_array() {
+                                        if property[0].as_str().unwrap_or("") == "tel" {
+                                            let abuse_contact_phone =
+                                                property[3].as_str().unwrap_or("").to_string();
+                                            if !abuse_contact_phone.is_empty() {
+                                                front_formatted_data.push_str(&format!(
+                                                    "Registrar Abuse Contact Phone: {}\n",
+                                                    abuse_contact_phone
+                                                ));
+                                            }
+                                        } else if property[0].as_str().unwrap_or("") == "email" {
+                                            let abuse_contact_email =
+                                                property[3].as_str().unwrap_or("").to_string();
+                                            if !abuse_contact_email.is_empty() {
+                                                front_formatted_data.push_str(&format!(
+                                                    "Registrar Abuse Contact Email: {}\n",
+                                                    abuse_contact_email
+                                                ));
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+// Where do we move this to?
+// capitalize first letter
```

Comment:
Unless you start collecting a lot of these string utilities, just leave it here. There is a `strings.rs` under `md` but that's because it required a lot of special cases.